batcher/aligned-sdk/src/core/types.rs

```diff
@@ -170,14 +186,62 @@ pub struct ClientMessage {
     pub signature: Signature,
 }
 
+impl Eip712 for NoncedVerificationData {
+    type Error = Eip712Error;
+
+    fn domain(&self) -> Result<EIP712Domain, Self::Error> {
+        let payment_service_addr = get_payment_service_addr(self.chain_id);
+
+        Ok(EIP712Domain {
+            name: Some("Aligned".into()),
+            version: Some("1".into()),
+            chain_id: Some(self.chain_id),
+            verifying_contract: payment_service_addr,
+            salt: None,
+        })
+    }
+
+    fn type_hash() -> Result<[u8; 32], Self::Error> {
+        let mut hasher = Keccak256::new();
+        hasher.update(NONCED_VERIFICATION_DATA_TYPE);
+        Ok(hasher.finalize().into())
+    }
+
+    fn struct_hash(&self) -> Result<[u8; 32], Self::Error> {
+        let verification_data_hash =
+            VerificationCommitmentBatch::hash_data(&self.verification_data.clone().into());
+
+        let mut hasher = Keccak256::new();
+
+        hasher.update(NONCED_VERIFICATION_DATA_TYPE);
+        let nonced_verification_data_type_hash = hasher.finalize_reset();
+
+        hasher.update(self.nonce);
+        let nonce_hash = hasher.finalize_reset();
+
+        hasher.update(
+            [
+                nonced_verification_data_type_hash.as_slice(),
+                verification_data_hash.as_slice(),
+                nonce_hash.as_slice(),
+            ]
+            .concat(),
+        );
```

Comment:
check if this is equivalent to calling hasher.update with each slice instead of using concat:

```suggestion
        hasher.update(nonced_verification_data_type_hash.as_slice());
        hasher.update(verification_data_hash.as_slice());
        hasher.update(nonce_hash.as_slice());
```

Reply:
Yes, it is equivalent. Thanks for point it out, solved in [f259a96](https://github.com/yetanotherco/aligned_layer/pull/916/commits/f259a96e4707d809fb89e5d5d6601e82e9668c17)